packages/frontend/src/components/Explorer/SpaceBrowser/index.tsx

```diff
@@ -53,11 +51,19 @@ const SpaceBrowser: FC<{ projectUuid: string }> = ({ projectUuid }) => {
                                     <FolderIcon icon="folder-close"></FolderIcon>
                                 </FolderWrapper>
                                 <SpaceTitle>{name}</SpaceTitle>
-                                <SpaceBrowserMenu
-                                    spaceUuid={uuid}
-                                    onDeleteSpace={setDeleteSpaceUuid}
-                                    onUpdateSpace={setUpdateSpaceUuid}
-                                />
+                                <div
+                                    onClick={(e) => {
+                                        // prevent clicks in menu to trigger redirect
+                                        e.stopPropagation();
+                                        e.preventDefault();
```

Comment:
As `<div>` tag has no default behavior ` e.preventDefault();`, so you can delete this one :) 

```suggestion
```

Reply:
If I remove that line the click even of the children will still trigger the redirect.

Reply:
that will handle by `e.stopPropagation();` basically :)